From aefbd8a078847ef1e1076b16514c5168c5fc9858 Mon Sep 17 00:00:00 2001
Message-Id: <aefbd8a078847ef1e1076b16514c5168c5fc9858.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Pallavi <pallavi.g@intel.com>
Date: Wed, 12 Nov 2014 10:56:03 +0530
Subject: [PATCH 73/75] MUST_REBASE [VPG]: adf/intel/vlv: fix the src rec
 fixed pt check issue

Considered the src_x and src_y zero scenario also as part
of the src rec format check.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need
to work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: Ifdc22eb8ee8688a64ebc1e8d0ddd83e2d3f615ba
Signed-off-by: Pallavi <pallavi.g@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_sp_plane.c |   16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c b/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c
index 7ef7c39..3d08777 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c
@@ -411,10 +411,18 @@ static int vlv_sp_validate(struct intel_plane *plane, struct intel_buffer *buf,
 	};
 
 	/* make sure the src rectangle in 16.16 fixed point format */
-	if (!(config->src_x / (1<<16)) ||
-	    !(config->src_w / (1<<16)) ||
-	    !(config->src_y / (1<<16)) ||
-	    !(config->src_h / (1<<16))) {
+	if (!(config->src_w / (1 << 16)) ||
+	    !(config->src_h / (1 << 16))) {
+		pr_err("ADF:src rec are not in 16.16 fixed fmt%s\n", __func__);
+		return -ERANGE;
+	}
+
+	if (config->src_x && !(config->src_x / (1 << 16))) {
+		pr_err("ADF:src rec are not in 16.16 fixed fmt%s\n", __func__);
+		return -ERANGE;
+	}
+
+	if (config->src_y && !(config->src_y / (1 << 16))) {
 		pr_err("ADF:src rec are not in 16.16 fixed fmt%s\n", __func__);
 		return -ERANGE;
 	}
-- 
1.7.9.5

